src/test/java/org/prebid/server/auction/AmpRequestFactoryTest.java

```diff
@@ -361,6 +363,613 @@ public void shouldRespondWithBidRequestWithTestFlagOn() {
         assertThat(captor.getValue().getTest()).isEqualTo(1);
     }
 
+    @Test
+    public void shouldReturnBidRequestWithOverwrittenTagIdBySlotParamValue() {
+        // given
+        given(httpRequest.getParam("tag_id")).willReturn("tagId");
+        given(httpRequest.getParam("slot")).willReturn("overwritten-tagId");
+
+        final BidRequest bidRequest = givenBidRequest(
+                builder -> builder.ext(mapper.valueToTree(ExtBidRequest.of(null))),
+                Imp.builder().build());
+
+        given(storedRequestProcessor.processAmpRequest(anyString())).willReturn(Future.succeededFuture(bidRequest));
+        given(auctionRequestFactory.fillImplicitParameters(any(), eq(routingContext)))
+                .willAnswer(answerWithFirstArgument());
+        given(auctionRequestFactory.validateRequest(any())).willAnswer(answerWithFirstArgument());
+
+        // when
+        final Future<BidRequest> future = factory.fromRequest(routingContext);
+
+        // then
+        assertThat(future.succeeded()).isTrue();
+
+
+        assertThat(singletonList(future.result()))
+                .flatExtracting(BidRequest::getImp)
+                .extracting(Imp::getTagid)
+                .containsOnly("overwritten-tagId");
+    }
+
+    @Test
+    public void shouldReturnBidRequestWithOverwrittenSitePageByCurlParamValue() {
```

Comment:
m.b. we need also another test when `BidRequest.site` is null. What do you think?